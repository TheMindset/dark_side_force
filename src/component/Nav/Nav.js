import React from 'react'
import './Nav.scss'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom'

const Nav = ({ user }) => {
  const { name, quote, rank } = user
  return (
    <nav>
      <div className='user-profile'>
        <h1> {name} </h1>
        <p> "{quote}" </p>
        <h2> {rank} </h2>
      </div>

      <NavLink to='/movies' className='nav-link'>Movies</NavLink>
      <NavLink to='/'>
        <button>LogOut</button>
      </NavLink>
    </nav>
  )
}

export default Nav

Nav.propTypes = {
  user: PropTypes.object.isRequired
}