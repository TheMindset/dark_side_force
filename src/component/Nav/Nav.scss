nav {
  border: 1px solid black;
  display: flex;
  align-items: center;
  justify-content: space-between;

  div {
    display: flex;
    flex-direction: column;
  }

  button {
    &:hover {
      cursor: pointer;
    }
  }
}