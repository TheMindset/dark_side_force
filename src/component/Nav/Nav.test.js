import React from 'react'
import { shallow } from 'enzyme'
import Nav from './Nav'

describe('Nav', () => {
  let wrapper, mockUser, mockLogOut, mockNumFav

  beforeEach(() => {
    mockLogOut = jest.fn()
    mockNumFav = 3
    mockUser = {
      name: 'Hola',
      quote: 'The force of your mind',
      rank: 'Padawan'
    }
    wrapper = shallow(<Nav user={mockUser} logOut={mockLogOut} numFav={mockNumFav} />)
  })

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
