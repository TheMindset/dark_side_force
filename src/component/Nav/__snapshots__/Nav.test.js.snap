// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<nav>
  <div
    className="user-profile"
  >
    <div
      className="inner"
    >
      <span>
        i
      </span>
      <p>
        Name: 
        Hola
         
      </p>
      <p>
        Quote: "
        The force of your mind
        " 
      </p>
      <p>
        Rank :
        Padawan
         
      </p>
    </div>
  </div>
  <NavLink
    className="nav-link"
    to="/movies"
  >
    Movies
  </NavLink>
  <NavLink
    className="nav-link"
    to="/favorites"
  >
    Favorites (
    3
    )
  </NavLink>
  <Link
    className="nav-link"
    onClick={[MockFunction]}
    to="/"
  >
    <button>
      LogOut
    </button>
  </Link>
</nav>
`;
